src/Illuminate/Database/Query/Grammars/PostgresGrammar.php

```diff
@@ -20,6 +21,43 @@ class PostgresGrammar extends Grammar
         'is distinct from', 'is not distinct from',
     ];
 
+    /**
+     * {@inheritdoc}
+     *
+     * @param Builder $query
+     * @param array   $where
+     *
+     * @return string
+     */
+    protected function whereBasic(Builder $query, $where)
+    {
+        // Special treatment for like clauses. In order to make
+        // any column searchable, we need to convert it to text.
+        if (Str::contains($where['operator'], 'like')) {
```

Comment:
`strtolower($where['operator'])` to handle uppercase `LIKE`.

Reply:
implemented and added an extra assertion for this case